### PR TITLE
[monitoring-kubernetes-control-plane] Refactoring

### PR DESCRIPTION
**Pull request by nabokihms**
Signed-off-by: m.nabokikh <maksim.nabokikh@flant.com>

## Description
* Simplify the design of the module
* Substitute bash hooks with hooks written in Go
* Add openapi shema

## Why do we need it, and what problem does it solve?
Refactoring.

Closes https://github.com/deckhouse/deckhouse/issues/47

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [x] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: monitoring-kubernetes-control-plane
type: feature
summary: Entirely rework the module
impact: The monitoring-kubernetes-control-plane module now works only if the control-plane-manager module is enabled. 
impact_level: high
---
section: common
type: feature
summary: Update kube-rbac-proxy
impact: Prometheus, ingress-controller, cilium and other pods will be restarted.
impact_level: high
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @nabokihms. I've started the workflow for label 'e2e/run/gcp'...

File: modules/340-monitoring-kubernetes-control-plane/docs/CONFIGURATION.md [hunk 1]
```diff
@@ -3,221 +3,3 @@ title: "Monitoring the control plane: configuration"
 ---
 
 The `monitoring-kubernetes-control-plane` module is configured automatically and usually does not require manual configuring. However, manual configuring may be necessary in some non-standard cases.
```
Comment (z9r5) [inline modules/340-monitoring-kubernetes-control-plane/docs/CONFIGURATION.md:5]: > However, manual configuring may be necessary in some non-standard cases.

But there are no parameters at all.  Maybe it's better to delete this sentence?

**Review by z9r5 — DISMISSED** [review]

**Review by apolovov — APPROVED** [review]
Great job!

File: modules/340-monitoring-kubernetes-control-plane/docs/CONFIGURATION.md (hunk 1 above)
Comment (konstantin-axenov) [inline modules/340-monitoring-kubernetes-control-plane/docs/CONFIGURATION.md:5]:
```suggestion
The `monitoring-kubernetes-control-plane` module is configured automatically.
```

**Review by konstantin-axenov — APPROVED** [review]

**Pull request merged**